### ISSUE TITLE
GNS3 API blocks on invalid request methods

### ISSUE THREAD

**Issue by b-ehlers** [GNS3/gns3-server#1766]
GNS3 version 2.2.8 on Darwin (64-bit) with Python 3.7.6 Qt 5.13.2 and PyQt 5.13.2.

An API request with an invalid request method results in a blocked connection, no response (not even an error message) is returned.

Working:
```
/Users/behlers# curl -X GET -u 'admin:password' http://192.168.1.10:3080/v2/version
{
    "local": true,
    "version": "2.2.8"
}
```

Invalid method `GET1`:

```
/Users/behlers# curl -X GET1 -u 'admin:password' http://192.168.1.10:3080/v2/version
<no response, blocks until killed>
```

Here a wireshark trace, the GNS3 server is returning no data.
![wireshark](https://user-images.githubusercontent.com/49040325/83129860-b7cb6c00-a0dd-11ea-86cf-e45e4b895ce0.png)

**Comment by grossmj** [comment]
Looks like the problem is with aiohttp

I ran their simple server example: https://github.com/aio-libs/aiohttp/blob/master/examples/server_simple.py

Then tried `curl -X GET1 http://0.0.0.0:8080/` and I got the same result as you.

**Comment by b-ehlers** [comment]
For me this issue has a low priority. It happened, when I made a typing error using the GNS3 API. I can easily check the request method in my API library, which would prevent another occurance of this issue.

If it's simple to fix on the server side, that's nice. Otherwise document it as a known restriction.

**Comment by grossmj** [comment]
I posted the problem on the aiohttp repo. I will close this since there is nothing we can do on our side I think.

**Comment by b-ehlers** [comment]
Thanks for taking care of this issue.

As an easy fix, I have added a check for unvalid methods in my library. So I'm safe now.

**Comment by b-ehlers** [comment]
Fixed in aiohttp v3.7.0, see <https://github.com/aio-libs/aiohttp/releases/tag/v3.7.0>
